File: cases/taint/kotlin/local-overwrite-negative/LocalOverwriteNegative.kt
package dataflowbench

object LocalOverwriteNegative {
    fun dfb_source(): Int { // DFB-SOURCE: local-overwrite-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: local-overwrite-sink

    fun run() {
        var value = dfb_source()
        value = 0 // DFB-KILL: local-overwrite-clean
        dfb_sink(value)
    }
}

File: cases/taint/csharp/local-overwrite-negative/LocalOverwriteNegative.cs
namespace DataFlowBench;

static class LocalOverwriteNegative
{
    static int dfb_source() // DFB-SOURCE: local-overwrite-input
    {
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: local-overwrite-sink

    static void Run()
    {
        int value = dfb_source();
        value = 0; // DFB-KILL: local-overwrite-clean
        dfb_sink(value);
    }
}
